arctic/_compression.py

```diff
@@ -20,23 +24,27 @@ def enable_parallel_lz4(mode):
             True: Use parallel compression. False: Use sequential compression
     """
     global ENABLE_PARALLEL
-    ENABLE_PARALLEL = mode
+    ENABLE_PARALLEL = bool(mode)
     logger.info("Setting parallelisation mode to {}".format("multithread" if mode else "singlethread"))
 
 
-def compress_array(str_list):
+def compress_array(str_list, withHC=False):
     """
     Compress an array of strings
     """
-    if not ENABLE_PARALLEL:
-        return [clz4.compress(s) for s in str_list]
-
-    # Less than 50 chunks its quicker to compress sequentially..
-    if len(str_list) > LZ4_N_PARALLEL:
-        return clz4.compressarr(str_list)
-    else:
-        return [clz4.compress(s) for s in str_list]
-
+    if not str_list:
+        return str_list
+
+    if not ENABLE_PARALLEL or len(str_list) <= LZ4_N_PARALLEL:
+        # Less than 50 chunks its quicker to compress sequentially..
```

Comment:
If we change LZ4_N_PARALLEL this comment will become outdated. Maybe we can say "Parallel compression is only worth it when we have a lot of chunks" or something like that.

Reply:
yes, correct.
Once I find the sweet spot I will also change this comment